TaskLayer/SearchTask/PostSearchAnalysisTask.cs

```diff
@@ -905,6 +909,147 @@ private void WritePeptideResults()
+            string variantPsmFile = Path.Combine(Parameters.OutputFolder, "VariantPSMs.psmtsv");
+            string variantPeptideFile = Path.Combine(Parameters.OutputFolder, "VariantPeptides.psmtsv");            
+            List<PeptideSpectralMatch> FDRPsms = Parameters.AllPsms
+                .Where(p => p.FdrInfo.QValue <= CommonParameters.QValueOutputFilter
+                && p.FdrInfo.QValueNotch <= CommonParameters.QValueOutputFilter && p.BaseSequence != null).ToList();
+            List<PeptideSpectralMatch> possibleVariantPsms = new List<PeptideSpectralMatch>();
+            foreach (var peptide in FDRPsms)
+            {
+                var variants = peptide.BestMatchingPeptides.SelectMany(p => p.Peptide.Protein.AppliedSequenceVariations.Select(v => PsmTsvWriter.IntersectsWithVariation(p.Peptide, v, true)).ToList()).ToList();
+                if(variants.Contains(true))
+                {
+                        possibleVariantPsms.Add(peptide);                                        
+                }                
+            }
+            
+            if (!Parameters.SearchParameters.WriteDecoys)
+            {
+                possibleVariantPsms.RemoveAll(b => b.IsDecoy);
+            }
+            if (!Parameters.SearchParameters.WriteContaminants)
+            {
+                possibleVariantPsms.RemoveAll(b => b.IsContaminant);
+            }
+
+            WritePsmsToTsv(possibleVariantPsms, variantPsmFile, Parameters.SearchParameters.ModsToWriteSelection);
+
+            List<PeptideSpectralMatch> variantPeptides = possibleVariantPsms.GroupBy(b => b.FullSequence).Select(b => b.FirstOrDefault()).ToList();
+            List<PeptideSpectralMatch> confidentVariantPeps = new List<PeptideSpectralMatch>();
+            WritePsmsToTsv(variantPeptides,variantPeptideFile, Parameters.SearchParameters.ModsToWriteSelection);
+            variantPeptides.RemoveAll(p => p.FdrInfo.QValue > 0.01 || p.FdrInfo.QValueNotch > 0.01 || p.IsDecoy);
+            foreach (var entry in variantPeptides)
+            {
+                var pwsm = entry.BestMatchingPeptides;
+                var nonvariantProtein = false;
+                foreach (var pep in pwsm)
+                {
+                    if (pep.Peptide.Protein.NonVariantProtein.Name == pep.Peptide.Protein.Accession)
+                    {
+                        nonvariantProtein = true;
+                    }
+                }
+                if (nonvariantProtein != true)
+                {
+                    confidentVariantPeps.Add(entry);
+                }
+            }
+
+            int savCount = 0;
+            int insertionCount = 0;
+            int deletionCount = 0;
+            int frameshiftCount = 0;
+            int stopGainCount = 0;
+            int stopLossCount = 0;
+            int exonLossCount = 0; ;
+            confidentVariantPeps.RemoveAll(p => p.FdrInfo.QValue > 0.01 || p.FdrInfo.QValueNotch > 0.01 || p.IsDecoy);            
```

Comment:
are we never reporting psms with high q-values

Reply:
I can include high q-value psms in output files but filter them out prior to results summary. I want the text results summary to only be 1% FDR.